libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -10,9 +17,21 @@ import { getReadingList, removeFromReadingList } from '@tmo/books/data-access';
 export class ReadingListComponent {
   readingList$ = this.store.select(getReadingList);
 
-  constructor(private readonly store: Store) {}
+  constructor(private readonly store: Store, private _snackBar: MatSnackBar) {}
 
   removeFromReadingList(item) {
     this.store.dispatch(removeFromReadingList({ item }));
+    const snackBarRef = this._snackBar.open('Removed', 'Undo', {
+      duration: 5000
+    });
+    snackBarRef.onAction().subscribe(() => {
+      this.store.dispatch(UndoremoveFromReadingList({ item }));
+    });
+  }
+
+  finishReadingList(item) {
+    this.store.dispatch(markedAsFinished({ item }));
+
+    this.store.dispatch(loadReadingList());
```

Comment:
Added code for showing snack bar functionality when adding and removing from a reading list